### PR TITLE
refactor dg and  add manual testing write up 

### PR DESCRIPTION
**Pull request by softmagnet**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -989,6 +997,61 @@ testers are expected to do more *exploratory* testing.
 
 2. _{ more test cases …​ }_
 
+### Editing a person
+
+1. Test case 1: Editing an existing student's own detail 
+   1. Prerequisites
+      2. Add student: `add n/John p/98765432 e/johnd@example.com a/311, Clementi Ave 2, #02-25 t/Chemistry t/Sec 3 nok/ n/Jack Doe p/10987654 e/jackd@example.com a/311, Clementi Ave 2, #02-25`
```
Comment (yourally2) [inline docs/DeveloperGuide.md:1004]: Should this be 1 instead of 2?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -989,6 +997,61 @@ testers are expected to do more *exploratory* testing.
...
+      3. Note the student and his or her index in the displayed list in the `Student` tab.
+   2. Edit the student (assuming the student's index is 1): `edit 1 n/updated student p/99993293 e/edited@gmail.com a/edited address t/`
+   3. Expected: All fields are changed according to `edit` command inputs and all tags are removed.
+2. Test case 2: Edit an existing student's NOK detail
+   1. Prerequisites: same as above
```
Comment (yourally2) [inline docs/DeveloperGuide.md:1009]: Capital letter for same? ie 'Same' and the rest as well

**Review by yourally2 — COMMENTED** [review]
LGTM!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (softmagnet) [reply docs/DeveloperGuide.md:1004]: good catch!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (softmagnet) [reply docs/DeveloperGuide.md:1009]: good catch!

**Comment by s7u4rt99** [comment]
LGTM

**Pull request merged**